Plugins/NativeServerPlugin/src/directx_buffer_capturer.cpp

```diff
@@ -0,0 +1,258 @@
+#include "pch.h"
+
+#include "directx_buffer_capturer.h"
+#include "plugindefs.h"
+
+#include "webrtc/modules/video_coding/codecs/h264/h264_encoder_impl.h"
+
+using namespace Microsoft::WRL;
+using namespace StreamingToolkit;
+
+DirectXBufferCapturer::DirectXBufferCapturer(ID3D11Device* d3d_device) :
+	d3d_device_(d3d_device)
+{
+}
+
+void DirectXBufferCapturer::Initialize(bool headless, int width, int height)
+{
+	// Gets the device context.
+	d3d_device_->GetImmediateContext(&d3d_context_);
+
+#ifdef MULTITHREAD_PROTECTION
+	// Enables multithread protection.
+	ID3D11Multithread* multithread;
+	d3d_device_->QueryInterface(IID_PPV_ARGS(&multithread));
+	multithread->SetMultithreadProtected(true);
+	multithread->Release();
+#endif // MULTITHREAD_PROTECTION
+
+	// Initializes NVIDIA encoder.
+	webrtc::H264EncoderImpl::SetDevice(d3d_device_.Get());
+	webrtc::H264EncoderImpl::SetContext(d3d_context_.Get());
+
+	// Headless mode initialization.
+	headless_ = headless;
+	if (headless_)
+	{
+		ResizeRenderTexture(width, height);
+	}
+}
+
+void DirectXBufferCapturer::SendFrame(int64_t prediction_time_stamp)
+{
+	if (!headless_)
```

Comment:
if this is code that we'd expect another developer to call i'd much prefer to see some error condition occur here instead of an early return...since it seems the caller should not call this signature unless they are headless. that way the dev can better diagnose what's going on

Reply:
Your understanding is correct. I can display an error dialog in non-headless mode.